### PR TITLE
Add stylelint to lint `*.scss` files per WordPress Coding Standards

### PR DESCRIPTION
**Pull request by ntwb**
First pass at adding stylelint to lint SCSS files per WordPress CSS Coding Standards

References:
• https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/
• https://github.com/stylelint/stylelint
• https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress

### CODE REVIEW

File: blocks/editable/style.scss [hunk 1]
```diff
@@ -3,6 +3,7 @@
 }
 
 .blocks-editable__tinymce {
+
```
Comment (aduth) [inline blocks/editable/style.scss]: What's the guideline that directs this blank line here? Throughout the changes I'm inferring a guideline of "always add a newline before nested selector". Is that documented somewhere? Looks a bit out-of-place in some cases, particularly in contrast to non-nested selector styles not needing the new line.

File: blocks/editable/style.scss [hunk 2]
```diff
@@ -117,6 +118,7 @@ input.editable-format-toolbar__link-input {
 	position: relative;
 
 	&:after {
-		@include long-content-fade( $size: 40% );
+
```
Comment (aduth) [inline blocks/editable/style.scss]: Similarly, what's the guideline for the newline here? Doesn't follow my inferred guideline from my previous comment.

File: docs/tinymce-per-block/src/assets/stylesheets/main.scss [hunk 3]
```diff
@@ -1,19 +1,19 @@
-@import 'variables';
```
Comment (aduth) [inline docs/tinymce-per-block/src/assets/stylesheets/main.scss]: I think we ought to exclude `docs` from these rules. They're more legacy prototypes than actively maintained.

Comment (ntwb) [inline docs/tinymce-per-block/src/assets/stylesheets/main.scss]: Right, shall do 👍

File: blocks/editable/style.scss (hunk 1 above)
Comment (ntwb) [inline blocks/editable/style.scss]: I agree that it does look out of place in places, it's been a tough one to solve

I've a [specific set of issues](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress/milestone/2) relating to _new lines_ that I need to run through, hopefully I can get to this in the next few days, I'll then refresh this PR an we can can go from there

File: blocks/editable/style.scss (hunk 2 above)
Comment (ntwb) [inline blocks/editable/style.scss]: See above

**Comment by ntwb** [comment]
Sorry for the delay in addressing your concerns @aduth, per the review comments above I hope to get to refresh this PR in the next few days 🤞

**Review by gziolo — CHANGES_REQUESTED** [review]
This one got a bit outdated :( Should we rebase and take another try to get it in?

**Comment by ntwb** [comment]
I'll create a fresh PR once I've update the `stylelint-config-wordpress` SCSS rules

**Pull request closed without merging**